df.buildings.xml

```diff
@@ -667,8 +667,32 @@
         <int32_t/>
         <int32_t/>
         <int32_t name='zone_num'/>
-        <int32_t name='dir_x' comment='dir_x and dir_y appear to be flags variables'/>
-        <int32_t name='dir_y'/>
+        <compound name='zone_settings' is-union='true'>
+            <compound name='whole'>
+                <int32_t name='i1'/>
+                <int32_t name='i2'/>
+            </compound>
```

Comment:
is there any reason this isn't a single uint64_t?

Reply:
I think it would change the alignment of the union and maybe its position.

Reply:
Alignment was the specific reason for this, but I may be incorrect:

The civzone struct looks like this prior to zone_settings

```std::vector<int32_t > assigned_units;
    std::vector<int32_t > assigned_items;
    df::civzone_type type; --int32, aligned to pointer
    int32_t is_active; --aligned to int32
    int32_t anon_1; --64
    int32_t anon_2; --32
    int32_t zone_num; -- 64
    T_zone_settings zone_settings; --only 4 byte aligned
```

So in my understanding, using a uint64_t would require 4 bytes of padding here to correctly align it to 8 bytes

Reply:
this is correct, an int64 would be aligned to a 64-bit boundary but a structure of two int32s is aligned to a 32-bit boundary